src/fluree/db/index.cljc

```diff
@@ -145,7 +145,8 @@
 
 (defn child-entry
   [{:keys [first] :as node}]
-  [first node])
+  (let [child-node (unresolve node)]
```

Comment:
I'm going to try to summarize my understanding of this to make sure I've got it correct:

`child-entry` takes a node and returns a tuple of the the first flake it addresses and the node itself. This change `unresolve`s the node itself so it won't carry around the `:children` or `:flakes` references.

Is it to reduce the size of the node we need to pass around?

Reply:
Yes. This code is mainly used when the database is re-indexed. In that case, the entire index tree is loaded into memory, node by node, and rebalanced. Without this change, the result of the indexing process would hold the all the data from an index in memory, thus eliminating the gains from processing the index node by node instead of all at once. This change instead keeps a pointer to the top level child nodes on disk, and those child nodes keep pointers to their children, and so on. The whole benefit of this change is memory efficiency for the indexing process.

Reply:
Nice! Thanks for the explanation, that makes perfect sense.